.github/workflows/ci.yml

```diff
@@ -33,6 +41,7 @@ jobs:
       - run: |
           ./scripts/codegen.sh
           [[ -z $(git status -s) ]] # Fail if changed. See https://stackoverflow.com/a/9393642
+        if: matrix.os == 'ubuntu-latest'
```

Comment:
The `scripts/codegen.sh` step can't run on Windows for a few reasons:
* Even setting this as `shell: bash`, the `wget` command doesn't quite work.
* Even rewriting the `script/download-model.sh` for Windows compatibility, the generated files are different due to `\n` vs `\r\n` line ending differences.

As such, I've opted to restrict the codegen-unchanged step to just run on Ubuntu.

A proper fix would involve changing the codegen logic.

I think the culprit is the use of `writeln!` (rather than `write!(concat!(..., "\n"), ...)`) in the `macro_rules! g { .. }` in `codegen-writer`'s `lib.rs`.

Reply:
According to the docs, [`writeln!`](https://doc.rust-lang.org/std/macro.writeln.html) always emits `\n`. It may be related to `git` settings.

Reply:
Ah, it must have been the other way 'round then and git check out code with `\r\n`. It doesn't matter :-)